opentracing/src/main/scala/zio/telemetry/opentracing/OpenTracing.scala

```diff
@@ -103,10 +103,9 @@ object OpenTracing {
   ): ZIO[R1, E, Span] =
     ZIO.service[OpenTracing.Service].flatMap { service =>
       Task(service.tracer.extract(format, carrier))
-        .fold(_ => None, Option.apply)
-        .flatMap {
-          case None          => zio
-          case Some(spanCtx) =>
+        .foldM(
+          _ => zio,
+          spanCtx =>
             for {
               current <- service.currentSpan.get
               span    <- UIO(service.tracer.buildSpan(operation).asChildOf(spanCtx).start())
```

Comment:
Please pardon my lurking, but I've been playing around with this library, and tracing in general. I too ran into this same problem, so this PR interested me. 

According to the documentation: https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/Tracer.java#L138-L152 if you pass in a `null` `SpanContext` or `Span`, it is a no op.

Do you think passing in a create function into the `spanFrom` would help?

Reply:
The `childOf` would be a no-op, which is what we indeed want. Without the PR, even `buildSpan` wouldn't be called if `spanCtx` was null. Does this help or did I not understand your question at all? 😄

Reply:
Ah, looking at this with fresher eyes, I see the intention.  This was me just getting tripped up on ZIO code which I'm still learning.  Looks much better now!